_test/general.test.php

```diff
@@ -35,27 +95,364 @@ public function test_plugininfo() {
+     * @expectedException InvalidArgumentException
+     */
+    public function test_exception_is_thrown_when_call_export_with_start_value_one($start, $count)
+    {
+        try {
+            $this->savePages('demopage1');
+
+            $xml = $this->getXML($start, $count);
+            $this->fail('An exception should be thrown when trying to call the export with corner case start and count values.');
+        } catch (\InvalidArgumentException $e) {
+            throw new \InvalidArgumentException($e);
+        }
+    }
+
+    public function parameterProviderForXMLCall()
+    {
+        return [
+            'test start = 1 and count = 1' => [1, 1],
+            'test start = 2 and count = 1' => [2, 1],
+            'test start = 0 and count = 0' => [0, 0],
+            'test start = 1 and count = 0' => [1, 0],
+            'test start = 0 and count = -1' => [0, -1],
+            'test start = -1 and count = 0' => [-1, 0]
+        ];
+    }
+
+    /**
+     * Test to ensure that parameters are correctly handled in the export call.
+     * start = 0, count = 10, total = 9;
+     */
+    public function test_call_export_with_count_greater_than_total()
+    {
+        $pageId = array();
+        for ($i = 1; $i <= 9; $i++) {
+            $pageId[$i] = 'demopage0' . $i;
+            $this->savePages($pageId[$i]);
+        }
+
+        $xml = $this->getXML();
+
+        $start = implode('', ($xml->xpath('/findologic/items/@start')));
+        $count = implode('', ($xml->xpath('/findologic/items/@count')));
+        $total = implode('', ($xml->xpath('/findologic/items/@total')));
+
+        $expectedStart = 0;
+        $expectedCount = $expectedTotal = 9;
+
+        $this->assertEquals($expectedStart, $start, 'Expected start value should match "0" when export is called with start "0".');
+        $this->assertEquals($expectedCount, $count, 'Expected count value should match "9" when export is called with "10" and $total is "9".');
+        $this->assertEquals($expectedTotal, $total, 'Expected total value should match "9" when DokuWiki has 9 pages.');
+    }
+
+    //======================================================================
+    // PLUGIN SETTING TESTS
+    //======================================================================
+
+    /**
+     * Test to ensure that the configuration works as expected and excluding of pages work properly.
+     * Correct page needs to be exported when 2 of 3 pages are excluded.
+     */
+    public function test_plugin_setting_exclude_pages()
+    {
+        $pageIds = array('settingtest1337', 'excludeme1', 'excludeme2');
+        $this->savePages($pageIds);
+
+        $conf = array();
+        // Set configuration
+        $conf['plugin']['findologicxmlexport']['excludePages'] = 'excludeme1, excludeme2';
+
+        $xml = $this->getXML(0, 20, $conf);
+
+        $total = implode('', ($xml->xpath('/findologic/items/@total')));
+        $secondItem = $xml->xpath('//item[@id="1"]');
```

Comment:
What if, due to a fascinating bug, the second page is assigned ID 2?

Reply:
Well.. my solution wasn't the best if I am honest. I am now counting the amount of items as I should have done before.